src/Bicep.MSBuild.E2eTests/src/example.ts

```diff
@@ -0,0 +1,72 @@
+
+import * as spawn from 'cross-spawn';
+import * as path from "path";
+import * as fs from "fs";
+import { SpawnSyncReturns } from 'node:child_process';
+
+export class Example {
+  readonly projectDir: string;
+  readonly projectFile: string;
+
+  constructor(projectName: string) {
+    const projectDir = path.normalize(path.join(__dirname, `../examples/${projectName}/`))
+      this.projectDir = projectDir,
+      this.projectFile = path.join(projectDir, `${projectName}.proj`)
+  }
+
+  public clean() {
+    const result = spawn.sync("git", ["clean", "-dfx", "--", this.projectDir], {
+      cwd: this.projectDir,
+      stdio: "pipe",
+      encoding: "utf-8",
+    });
+
+    if(result.status != 0) {
+      this.handleFailure("git", result);
+    }
+  }
```

Comment:
We had an issue raised about having an implicit dependency on git in our build process; might be nice to avoid this if possible - for example by staging files to a temp directory.

https://www.npmjs.com/package/nodegit is a potential solution, but I personally hate it - on some platforms it has to compile git from source which takes AGES

Reply:
Yeah I remember that, but it's rather difficult to get all the dependencies lined up to even run these tests correctly. We can fix once we get feedback.